src/snowflake/snowpark/modin/pandas/groupby.py

```diff
@@ -736,8 +736,30 @@ def all(self, skipna=True):
         ErrorMessage.method_not_implemented_error(name="all", class_="GroupBy")
 
     def size(self):
-        # TODO: SNOW-1063349: Modin upgrade - modin.pandas.groupby.DataFrameGroupBy functions
-        ErrorMessage.method_not_implemented_error(name="size", class_="GroupBy")
+        if self._axis == 1:
+            return DataFrameGroupBy(
+                self._df.T.iloc[:, [0]],
+                self._by,
+                0,
+                drop=self._drop,
+                idx_name=self._idx_name,
+                **self._kwargs,
+            ).size()
+        result = self._wrap_aggregation(
+            type(self._query_compiler).groupby_size,
+            numeric_only=False,
+        )
+        if not isinstance(result, Series):
+            result = result.squeeze(axis=1)
+        if not self._kwargs.get("as_index") and not isinstance(result, Series):
```

Comment:
so are we checking for the case where result is a dataframe and as index is true? maybe use positive check to make things more clear?

Reply:
All the code here was copied from Modin OSS. Syntax changes like this I would prefer to hold off on until they are changed in Modin first.

Reply:
is there a ticket for this? if we do not have ticket to track this, it is just going to be forget.

Reply:
SNOW-1063349 is how we are tracking the groupby methods aligning with Modin. This will get handled when we align with Modin.

Reply:
Would you be able to open the ticket on Modin with what you propose? I'm not sure I fully understand your question.

Reply:
Oh no, all i am proposing is just do check like isinstance(result, DataFrame),instead of not isinstance(result, Series), this just make it clear that I am handling dataframe case, or are we also intend to handle other case?

Reply:
This is where to file the issue in Modin: https://github.com/modin-project/modin/issues/new. Since it's your suggestion, I think it's best if you file it.